### PR TITLE
Translate "Ruby 3.3.0 released" (ko)

### PR DESCRIPTION
**Pull request by riseshia**
🔗 https://github.com/ruby/www.ruby-lang.org/issues/2818

Actual diff d81ce3f6875d33949c63a7eedede1d37e9870ebd

### CODE REVIEW

File: ko/news/_posts/2023-12-25-ruby-3-3-0-released.md [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+## Bison을 Lrama로 대체
+
+* Bison 대신 [Lrama LALR 파서 생성기](https://github.com/yui-knk/lrama)를 사용하도록 변경했습니다. [Feature #19637](https://bugs.ruby-lang.org/issues/19637)
+  * 관심이 있다면 [Ruby 파서의 미래 비전](https://rubykaigi.org/2023/presentations/spikeolaf.html)을 참고하세요.
+  * 유지보수성을 위해 Lrama 내부 구문 분석기가 Racc에서 생성한 LR 구문 분석기로 대체되었습니다.
+  * 매개변수화 규칙 `(?, *, +)`를 지원하며, Ruby의 parse.y에서 사용될 예정입니다.
+
+## YJIT
+
+* 3.2 버전 대비 주요 성능 개선 사항
```
Comment (yous) [inline ko/news/_posts/2023-12-25-ruby-3-3-0-released.md:36]:
```suggestion
* Ruby 3.2 버전 대비 주요 성능 개선 사항
```

File: ko/news/_posts/2023-12-25-ruby-3-3-0-released.md [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+  * 스플랫과 나머지 인자 지원이 개선되었습니다.
+  * 가상 머신의 스택 연산을 위해 레지스터가 할당됩니다.
+  * 선택적 인수가 포함된 더 많은 호출이 컴파일됩니다. 예외 처리기도 컴파일됩니다.
+  * 지원되지 않는 호출 타입인 경우, 클래스의 인스턴스 변수가 가지는 객체 형상의 조합이 매우 복잡한 경우에도
+    컴파일되기 전의 바이트 코드로 전환되지 않습니다.
+  * Rails의 `#blank?`와 [특수화한 `#present?`](https://github.com/rails/rails/pull/49909)는
+    인라인으로 최적화됩니다.
```
Comment (yous) [inline ko/news/_posts/2023-12-25-ruby-3-3-0-released.md:43]:
```suggestion
  * Rails의 `#blank?`와 [특수화한 `#present?`](https://github.com/rails/rails/pull/49909) 같은 단순한 메서드는
    인라인으로 최적화됩니다.
```

File: ko/news/_posts/2023-12-25-ruby-3-3-0-released.md [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+  * `Integer#!=`, `String#!=`, `Kernel#block_given?`, `Kernel#is_a?`,
+    `Kernel#instance_of?`, `Module#===`은 특별히 최적화되었습니다.
```
Comment (yous) [inline ko/news/_posts/2023-12-25-ruby-3-3-0-released.md:45]:
```suggestion
  * `Integer#*`, `Integer#!=`, `String#!=`, `String#getbyte`,
    `Kernel#block_given?`, `Kernel#is_a?`, `Kernel#instance_of?`, `Module#===`은 특별히 최적화되었습니다.
```

File: ko/news/_posts/2023-12-25-ruby-3-3-0-released.md [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+  * 3.2보다 컴파일 속도가 약간 개선되었습니다.
```
Comment (yous) [inline ko/news/_posts/2023-12-25-ruby-3-3-0-released.md:46]:
```suggestion
  * Ruby 3.2보다 컴파일 속도가 약간 개선되었습니다.
```

File: ko/news/_posts/2023-12-25-ruby-3-3-0-released.md [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+  * 이제 optcarrot에서 인터프리터보다 3배 이상 빠릅니다!
+* 3.2보다 메모리 사용량이 크게 개선되었습니다.
```
Comment (yous) [inline ko/news/_posts/2023-12-25-ruby-3-3-0-released.md:48]:
```suggestion
* Ruby 3.2보다 메모리 사용량이 크게 개선되었습니다.
```

File: ko/news/_posts/2023-12-25-ruby-3-3-0-released.md [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+* 코드 GC는 이제 비활성화된 상태로 시작합니다.
+  * `--yjit-exec-mem-size`를 새 코드 컴파일을 멈추는 최대 상한으로 취급합니다.
+  * 코드 GC로 인한 급격한 성능 하락이 사라집니다.
+    [Pitchfork](https://github.com/shopify/pitchfork)로 리포크를 사용하는 서버에서
+    더 나은 쓰기 시 복사 동작을 생성할 수 있습니다.
+  * 필요하다면 `--yjit-code-gc`로 코드 GC를 활성화할 수 있습니다.
+* 실행 시간에 YJIT을 활성화할 수 있는 `RubyVM::YJIT.enable`이 추가됩니다.
+  * 실행 명령의 인수나 환경 변수의 변경 없이 YJIT을 실행할 수 있습니다.
+    Rails 7.2는 이 메서드를 사용해
+    [기본으로 YJIT을 활성화](https://github.com/rails/rails/pull/49947)합니다.
```
Comment (yous) [inline ko/news/_posts/2023-12-25-ruby-3-3-0-released.md:63]:
```suggestion
    [기본으로 YJIT을 활성화](https://github.com/rails/rails/pull/49947)할 예정입니다.
```

File: ko/news/_posts/2023-12-25-ruby-3-3-0-released.md [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+  * 애플리케이션 부팅이 완료된 후에 YJIT를 활성화하는 데에도 사용할 수 있습니다.
+    부팅 시 YJIT를 비활성화한 상태에서 다른 YJIT 옵션을 사용하려는 경우
+    `--yjit-disable`을 사용할 수 있습니다.
+* 기본으로 더 많은 YJIT 통계 정보를 제공합니다.
+  * `yjit_alloc_size`와 메타 정보에 관련된 몇몇 통계가 기본으로 이용 가능합니다.
+  * `--yjit-stats`로부터 생성된 `ratio_in_yjit` 통계는 릴리스 빌드에서도 이용 가능하며,
+    대부분의 통계에 접근하는 데 특수한 통계나 개발 빌드는 더 이상 필요하지 않습니다.
+* 프로파일일 수단의 추가
```
Comment (yous) [inline ko/news/_posts/2023-12-25-ruby-3-3-0-released.md:71]:
```suggestion
* 프로파일링 기능 추가
```

File: ko/news/_posts/2023-12-25-ruby-3-3-0-released.md [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+* 보다 철저한 테스트와 여러 버그 수정
+
+## RJIT
+
+* 순수 Ruby JIT 컴파일러 RJIT을 도입하고 MJIT을 대체했습니다.
+  * RJIT은 Unix 플랫폼에서 x86-64 아키텍처만 지원합니다.
+  * MJIT과 달리 런타임에 C 컴파일러가 필요하지 않습니다.
+* RJIT은 실험 목적으로 존재합니다.
+  * 프로덕션 환경에서는 YJIT을 계속 사용해야 합니다.
+* Ruby용 JIT 개발에 관심이 있다면 [RubyKaigi 2023 3일 차에 있었던 k0kubun의 발표](https://rubykaigi.org/2023/presentations/k0kubun.html#day3)를 확인하세요.
```
Comment (yous) [inline ko/news/_posts/2023-12-25-ruby-3-3-0-released.md:83]:
```suggestion
* Ruby용 JIT 개발에 관심이 있다면 [RubyKaigi 2023 3일차에 있었던 k0kubun의 발표](https://rubykaigi.org/2023/presentations/k0kubun.html#day3)를 확인하세요.
```

File: ko/news/_posts/2023-12-25-ruby-3-3-0-released.md [hunk 9]
```diff
@@ -0,0 +1,304 @@
...
+
+## M:N thread scheduler
```
Comment (yous) [inline ko/news/_posts/2023-12-25-ruby-3-3-0-released.md:85]:
```suggestion
## M:N 스레드 스케줄러
```

File: ko/news/_posts/2023-12-25-ruby-3-3-0-released.md [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+
+* M:N 스레드 스케쥴러가 추가됩니다. [[Feature #19842]](https://bugs.ruby-lang.org/issues/19842)
```
Comment (yous) [inline ko/news/_posts/2023-12-25-ruby-3-3-0-released.md:87]:
```suggestion
* M:N 스레드 스케줄러가 추가됩니다. [[Feature #19842]](https://bugs.ruby-lang.org/issues/19842)
```

File: ko/news/_posts/2023-12-25-ruby-3-3-0-released.md [hunk 11]
```diff
@@ -0,0 +1,304 @@
...
+  * M개의 Ruby 스레드를 N개의 네이티브 스레드(OS 스레드)로 관리하므로 스레드의 생성, 관리 비용이 절감됩니다.
+  * C 확장 호환성에 문제가 발생할 가능성이 있으므로, 주 Ractor에서의 기본 설정은 비활성입니다.
+      * `RUBY_MN_THREADS=1` 환경 변수를 사용해 주 Ractor에서 M:N 스레드를 활성화할 수 있습니다.
+      * M:N 스레드는 주 Ractor가 아닌 Ractor에서 활성화되어 있습니다.
```
Comment (yous) [inline ko/news/_posts/2023-12-25-ruby-3-3-0-released.md:91]:
```suggestion
      * M:N 스레드는 주 Ractor가 아닌 Ractor에서 항상 활성화되어 있습니다.
```

File: ko/news/_posts/2023-12-25-ruby-3-3-0-released.md [hunk 12]
```diff
@@ -0,0 +1,304 @@
...
+  * `RUBY_MAX_CPU=n` 환경 변수는 `N`(네이티브 스레드의 최대 개수)의 최댓값을 지정합니다. 기본값은 8입니다.
+      * 하나의 Ractor에서는 하나의 Ruby 스레드만 실행 가능하므로 실제로 사용되는 네이티브 스레드의 개수는 `RUBY_MAX_CPU`에서 지정한 숫자나 실행 중인 Ractor의 개수보다 작습니다. 그러므로 (대다수를 차지하는) 단일 Ractor 애플리케이션에서는 하나의 네이티브 스레드를 사용합니다.
```
Comment (yous) [inline ko/news/_posts/2023-12-25-ruby-3-3-0-released.md:93]:
```suggestion
      * 하나의 Ractor에서는 하나의 Ruby 스레드만 실행 가능하므로 실제로 사용되는 네이티브 스레드의 개수는 `RUBY_MAX_CPU`에서 지정한 숫자나 실행 중인 Ractor의 개수보다 작습니다. 그러므로 (대다수를 차지하는) 단일 Ractor 애플리케이션에서는 단 하나의 네이티브 스레드를 사용합니다.
```

File: ko/news/_posts/2023-12-25-ruby-3-3-0-released.md [hunk 13]
```diff
@@ -0,0 +1,304 @@
...
+      * 블로킹 동작을 처리하기 위해, `N`개 이상의 네이티브 스레드가 사용될 수 있습니다.
+
+## 성능 향상
+
+* `defined?(@ivar)`가 객체 형상에 맞게 최적화됩니다.
+* `Socket.getaddrinfo`와 같은 이름 해결을 (POSIX 스레드가 사용 가능한 환경에서) 중단할 수 있게 됩니다. [Feature #19965](https://bugs.ruby-lang.org/issues/19965)
```
Comment (yous) [inline ko/news/_posts/2023-12-25-ruby-3-3-0-released.md:99]:
```suggestion
* `Socket.getaddrinfo`와 같은 이름 해결을 (POSIX 스레드가 사용 가능한 환경에서) 중단할 수 있게 됩니다. [[Feature #19965]](https://bugs.ruby-lang.org/issues/19965)
```

File: ko/news/_posts/2023-12-25-ruby-3-3-0-released.md [hunk 14]
```diff
@@ -0,0 +1,304 @@
...
+* 가비지 컬렉터의 몇몇 성능 향상
+  * 오래된 객체가 참조하는 어린 객체는 이제 즉시 오래된 객체로 마킹되지 않습니다.
+      이 동작은 메이저 GC의 빈도를 상당히 낮춰줍니다.
+      [[Feature #19678]](https://bugs.ruby-lang.org/issues/19678)
+  * 메이저 GC를 야기하는 보호받지 않은 객체의 숫자를 제어하는
+      새로운 변수 `REMEMBERED_WB_UNPROTECTED_OBJECTS_LIMIT_RATIO`가 추가되었습니다.
+      기본값은 `0.01` (1%)입니다. 이는 메이저 GC의 빈도를 상당히 낮춰줍니다.
```
Comment (yous) [inline ko/news/_posts/2023-12-25-ruby-3-3-0-released.md:106]:
```suggestion
      기본값은 `0.01`(1%)입니다. 이는 메이저 GC의 빈도를 상당히 낮춰줍니다.
```

File: ko/news/_posts/2023-12-25-ruby-3-3-0-released.md [hunk 15]
```diff
@@ -0,0 +1,304 @@
...
+      [[Feature #19571]](https://bugs.ruby-lang.org/issues/19571)
+  * `Time`, `Enumerator`, `MatchData`, `Method`, `File::Stat`, `BigDecimal` 등의
+      주요 타입에 부족했던 쓰기 보호가 구현되었습니다.
+      이는 마이너 GC에 걸리는 시간과 메이저 GC 빈도를 상당히 줄여줍니다.
+  * `Hash`, `Time`, `Thread::Backtrace`, `Thread::Backtrace::Location`,
+      `File::Stat`, `Method` 등의 대부분의 주요 클래스는 이제 가변 폭 할당을 사용합니다.
+      이는 해당 클래스의 할당 및 할당 해제 속도를 빠르게 만들며, 메모리 사용량과
+      힙 파편화를 줄여줍니다.
+  * 가비지 컬렉터에 약한 참조가 추가됩니다. [Feature #19783](https://bugs.ruby-lang.org/issues/19783)
```
Comment (yous) [inline ko/news/_posts/2023-12-25-ruby-3-3-0-released.md:115]:
```suggestion
  * 가비지 컬렉터에 약한 참조가 추가됩니다. [[Feature #19783]](https://bugs.ruby-lang.org/issues/19783)
```

File: ko/news/_posts/2023-12-25-ruby-3-3-0-released.md [hunk 16]
```diff
@@ -0,0 +1,304 @@
...
+
+또한, IRB는 광범위한 리팩토링을 거쳤으며, 향후 개선이 용이하도록 많은 버그 수정이 이루어졌습니다.
+
+## 호환성 문제
+
+주의: 기능 버그 수정은 포함되어 있지 않습니다.
+
+* 블록에서 매개변수 없이 `it`을 호출하는 기능은 삭제 예정입니다.
+  Ruby 3.4부터 `it`은 첫 번째 블록 인수를 가리킵니다.
+  [Feature #18980](https://bugs.ruby-lang.org/issues/18980)
```
Comment (yous) [inline ko/news/_posts/2023-12-25-ruby-3-3-0-released.md:138]:
```suggestion
  [[Feature #18980]](https://bugs.ruby-lang.org/issues/18980)
```

File: ko/news/_posts/2023-12-25-ruby-3-3-0-released.md [hunk 17]
```diff
@@ -0,0 +1,304 @@
...
+
+### 삭제된 환경 변수
+
+폐기 예정이었던 환경 변수가 삭제됩니다.
+
+* 환경 변수 `RUBY_GC_HEAP_INIT_SLOTS`는 폐기 예정이었으며 아무 효과도 없습니다. 대신 환경 변수 `RUBY_GC_HEAP_{0,1,2,3,4}_INIT_SLOTS`를 사용해 주세요. [Feature #19785](https://bugs.ruby-lang.org/issues/19785)
```
Comment (yous) [inline ko/news/_posts/2023-12-25-ruby-3-3-0-released.md:144]:
```suggestion
* 환경 변수 `RUBY_GC_HEAP_INIT_SLOTS`는 폐기 예정이었으며 아무 효과도 없습니다. 대신 환경 변수 `RUBY_GC_HEAP_{0,1,2,3,4}_INIT_SLOTS`를 사용해 주세요. [[Feature #19785]](https://bugs.ruby-lang.org/issues/19785)
```

File: ko/news/_posts/2023-12-25-ruby-3-3-0-released.md [hunk 18]
```diff
@@ -0,0 +1,304 @@
...
+## Stdlib 호환성 문제
+
+### `ext/readline` 폐기
+
+* 우리는 `ext/readline` API와 호환되는 순수 Ruby 구현인 `reline`을 가지고 있습니다. 앞으로는 `reline`에 의존할 것입니다. `ext/readline`을 사용해야 하는 경우, `gem install readline-ext`를 사용하여 rubygems.org를 통해 `ext/readline`을 설치할 수 있습니다.
+* 이제 더 이상 `libreadline` 또는 `libedit`과 같은 라이브러리를 설치할 필요가 없습니다.
+
+## 표준 라이브러리 갱신
+
+사용자가 미래의 Ruby 버전에서 내장될 예정의 gem을 직접 불러올 때 RubyGems와 Bundler가 경고 문구를 출력합니다.
```
Comment (yous) [inline ko/news/_posts/2023-12-25-ruby-3-3-0-released.md:155]:
```suggestion
사용자가 미래의 Ruby 버전에서 내장될 예정의 gem을 Gemfile이나 gemspec에 추가하지 않고 직접 `require` 할 때 RubyGems와 Bundler가 경고 문구를 출력합니다.
```

File: ko/news/_posts/2023-12-25-ruby-3-3-0-released.md [hunk 19]
```diff
@@ -0,0 +1,304 @@
...
+
+bootsnap gem을 사용하고 있다면, 해당 경고를 생략합니다. 적어도 한번 `DISABLE_BOOTSNAP=1` 환경 변수와 함께 애플리케이션을 실행하길 권장합니다. 이는 이번 버전에만 해당되는 제약입니다.
```
Comment (yous) [inline ko/news/_posts/2023-12-25-ruby-3-3-0-released.md:157]:
```suggestion
bootsnap gem을 사용하고 있다면, 해당 경고를 생략합니다. 적어도 한 번은 `DISABLE_BOOTSNAP=1` 환경 변수와 함께 애플리케이션을 실행하길 권장합니다. 이는 이번 버전에만 해당되는 제약입니다.
```

**Comment by riseshia** [comment]
Applied [b43f910](https://github.com/ruby/www.ruby-lang.org/pull/3162/commits/b43f910f6931872eed43b8fd6c63cb87481acf7e) 🙏

**Review by yous — APPROVED** [review]

**Pull request merged**